src/slider/slider.component.ts

```diff
@@ -147,38 +167,71 @@ export class Slider implements AfterViewInit, ControlValueAccessor {
 	/** Set the initial value. Available for two way binding */
 	@Input() set value(v) {
 		if (!v) {
-			v = this.min;
+			v = [this.min];
 		}
 
-		if (v > this.max) {
-			v = this.max;
+		if (typeof v === "number" || typeof v === "string") {
+			v = [Number(v)];
 		}
 
-		if (v < this.min) {
-			v = this.min;
+		if (v[0] < this.min) {
+			v[0] = this.min;
 		}
 
-		this._value = v;
+		if (v[0] > this.max) {
+			v[0] = this.max;
+		}
 
-		if (this.thumb) {
-			this.thumb.nativeElement.style.left = `${this.getFractionComplete(v) * 100}%`;
+		if (this.isRange()) {
+			if (this._previousValue[0] !== v[0]) { // left moved
+				if (v[0] > v[1] - this.step) {
+					// stop the left handle if surpassing the right one
+					v[0] = v[1] - this.step;
+				} else if (v[0] > this.max) {
+					v[0] = this.max;
+				} else if (v[0] < this.min) {
+					v[0] = this.min;
+				}
+			}
+
+			if (this._previousValue[1] !== v[1]) { // right moved
+				if (v[1] > this.max) {
+					v[1] = this.max;
+				} else if (v[1] < this._value[0] + this.step) {
+					// stop the right handle if surpassing the left one
+					v[1] = this._value[0] + this.step;
+				} else if (v[1] < this.min) {
+					v[1] = this.min;
+				}
+			}
 		}
 
-		if (this.filledTrack) {
-			this.filledTrack.nativeElement.style.transform = `translate(0%, -50%) ${this.scaleX(this.getFractionComplete(v))}`;
+		this._previousValue = [...this._value]; // store a copy, enable detection which handle moved
+		this._value = [...v]; // triggers change detection when ngModel value is an array (for range)
+
+		if (this.isRange() && this.filledTrack) {
+			this.updateTrackRangeWidth();
+		} else if (this.filledTrack) {
+			this.filledTrack.nativeElement.style.transform = `translate(0%, -50%) ${this.scaleX(this.getFractionComplete(v[0]))}`;
 		}
 
-		if (this.input) {
-			this.input.value = v.toString();
+		if (this.inputs && this.inputs.length) {
+			this.inputs.forEach((input, index) => {
+				input.value = this._value[index].toString();
+			});
 		}
 
 		this.propagateChange(v);
 		this.valueChange.emit(v);
```

Comment:
Right now these will always emit an array

```suggestion
		const valueToEmit = this.isRange() ? v : v[0];
		this.propagateChange(valueToEmit);
		this.valueChange.emit(valueToEmit);
```

Reply:
Thankfully this is the only place we call those!

Reply:
Ah yes, didn't come up in testing, good catch!